app/mailers/general_mailer.rb

```diff
@@ -1,18 +1,53 @@
 class GeneralMailer < ApplicationMailer
-  def reminder_email_to_non_participants(subject, content, sendDate)
+  def self.reminder_email_to_non_participants(sendDate, emailId)
     diffD = (sendDate - Date.today).to_i
-
-    Student.all.each do |student|
-      if student.has_participated == false and student.user
-          user = student.user
-          GeneralMailer.delay(queue:"reminder", run_at: diffD.days.from_now).reminder_email(user, subject, content)
-      end
-    end
+    GeneralMailer.delay(queue:"reminder", run_at: diffD.days.from_now).scheduled_email(emailId)
   end
 
-  def reminder_email(user, subject, content)
+  def self.reminder_email(user, subject, content)
     @user = user
     mail(to: user.email, subject: subject, body: content)
   end
 
+  def scheduled_email(emailId)
+    email = Email.find emailId
+    if email
```

Comment:
Should we also check to see if the date has been updated?

Reply:
We now just delete emails when the send date changes.